grpc/pb-marshalling.go

```diff
@@ -318,22 +359,27 @@ func PBToAuthz(pb *corepb.Authorization) (core.Authorization, error) {
 		}
 		challs[i] = chall
 	}
-	expires := time.Unix(0, pb.ExpiresNS).UTC()
+	var expires *time.Time
+	if pb.ExpiresNS != 0 {
+		c := time.Unix(0, pb.ExpiresNS).UTC()
+		expires = &c
+	}
 	authz := core.Authorization{
 		ID:             pb.Id,
 		Identifier:     identifier.ACMEIdentifier{Type: identifier.DNS, Value: pb.Identifier},
 		RegistrationID: pb.RegistrationID,
 		Status:         core.AcmeStatus(pb.Status),
-		Expires:        &expires,
+		Expires:        expires,
```

Comment:
Why is this change necessary / what benefit does it bring?

Reply:
When I was [adding tests](https://github.com/letsencrypt/boulder/pull/7070/files#diff-dd201678e04d27d421ec199e1320be8af836f5c71344ff66f7d8d4ab83ecc48fR254) for `<x>ToPB` and `PBto<x>` I wasn't getting the expected `nil` value back , and instead receiving a `time.Unix` zero value which is the Unix epoch.
```
--- FAIL: TestAuthz (0.00s)
    pb-marshalling_test.go:266: 
    [core.Authorization{ID:"1", Identifier:identifier.ACMEIdentifier{Type:"dns", Value:"example.com"}, RegistrationID:5, Status:"pending", Expires:<nil>, 
    !(deep)= 
    [core.Authorization{ID:"1", Identifier:identifier.ACMEIdentifier{Type:"dns", Value:"example.com"}, RegistrationID:5, Status:"pending", Expires:time.Date(1970, time.January, 1, 0, 0, 0, 0, time.UTC),
```